src/BatteryStateOfChargeService.h

```diff
@@ -18,5 +18,22 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
 
-    // Recommend adding necessary variables here
+    double InitialCurrent;
+    double CurrentPrime = 0;
+    double AmpHours;
+    double AmpHChange;
+    double AverageCurrent;
+    QTime InitialTime;
+    QTime TimePrime;
+    QTime t;
```

Comment:
It is very valuable to have variable names be more descriptive. It not only helps out someone else reading the code but from experience it also greatly helps when coming back to your own code. Also names like TimePrime are not very clear as I am not sure what Prime stands for within this context. I find naming one of the hardest things about programming but it does wonders at making readable clean code. Keep at it and put lots of thought into each variable name.